scarb/src/ops/resolve.rs

```diff
@@ -89,7 +91,12 @@ pub fn resolve_workspace(ws: &Workspace<'_>) -> Result<WorkspaceResolve> {
                 .map(|pkg| pkg.manifest.summary.clone())
                 .collect::<Vec<_>>();
 
-            let resolve = resolver::resolve(&members_summaries, &patched, ws.config().ui()).await?;
+            let lockfile: Lockfile = read_lockfile(ws)?;
```

Comment:
Shouldn't we ignore lockfile parsing errors? At least print them as warnings and overwrite the lockfile afterwards 🤔

Reply:
Hmm, I am not really sure. Ignoring parsing errors would be inconsistent from what Cargo does:
```
error: failed to parse lock file at: /Users/maciektr/Projects/cairo/Cargo.lock

Caused by:
  missing field `name`
  in `package`
```
So requiring an action from user (i.e. to remove / reset lock) may be intentional. 

On the other hand, npm seems to ignore invalid locks and overwrite without an error.

Reply:
Interesting. I suggested this exactly because I followed NPM logic. Didn't know that Cargo is erroring here. So let's stay with Cargo's behaviour, as we used to do. Thx.